app/support/products/scheduling_support.rb

```diff
@@ -94,12 +94,16 @@ def available_schedule_rules(user)
     end
   end
 
-  def online!
-    offline_reservations.current.update_all(reserve_end_at: Time.current)
+  def offline?
+    offline_reservations.current.any?
+  end
+
+  def offline_via_schedule_share?
```

Comment:
Is there a way we can hide this so the only public method is `offline?` and that checks the shared schedules as well. 

Thinking about it, I'm becoming less convinced that the shared schedule behavior we're implementing is correct, and would like to make it easy to change later.